### PR TITLE
Issue #41: Adding support for nodes without {id} being specified

### PR DESCRIPTION
**Pull request by ps23Rick**
Here's a snapshot for the moment for troubleshooting

### CODE REVIEW

File: headless.module [hunk 1]
```diff
@@ -84,32 +97,156 @@ function headless_menu() {
...
   if ($config[$type] != 1) {
     $json_error = ['code' => 404];
     backdrop_json_output($json_error);
     backdrop_exit();
   }
-  $my_json = node_load($nid);
-  backdrop_json_output($my_json);
-  backdrop_exit();
+  // load the requested node and check to see if types match or not
+  $requested_node = node_load($nid);
```
Comment (serundeputy) [inline headless.module]: we can't do this here, because we don't have an `$nid` passed in via URL this time.

File: headless.module [hunk 2]
```diff
@@ -84,32 +97,156 @@ function headless_menu() {
...
+  if ($requested_node->type == $type) {
```
Comment (serundeputy) [inline headless.module]: since we don't have `$requested_node` this if statement is moot.

**Comment by serundeputy** [comment]
@ps23Rick I left some comments.

In addition to the comments there are merge conflicts here ... I'm guessing you did not update your branch with the latest from the `1.x-1.x` branch.

**Comment by ps23Rick** [comment]
This is very much an in-work thing. I can guarantee there is stuff in there that just doesn't make sense.  I'm not done by any stretch.  You are correct that I didn't update my tree before doing my changes.

File: headless.module (hunk 1 above)
Comment (ps23Rick) [reply headless.module]: hmm.. Maybe I'm missing something or going about this the wrong way.  What I did was leave the existing headless_type() and headless_type_v2() methods alone -- the code above I didn't touch as part of this change -- that was associated with #38.. Git is not one of my strong suits and I'm not sure how to 'fix' things to ensure the PR is not showing stuff that I changed as part of the other PR.. Sorry!!!

Comment (ps23Rick) [reply headless.module]: I forgot to say that I added two new methods below the existing ones -- so there's 4 'type' methods now.  If you'd prefer them merged together I can do that I think..Maybe that would simplify things.. I don't know.

**Comment by serundeputy** [comment]
No problem; I think you are on the right track here.

We can either do a quick call or I can make a video about getting your fork up to date.

Then your changes are pretty close to getting the collection endpoints up

Let me know if you have 10 or 15min tomorrow

**Comment by ps23Rick** [comment]
Just to make things easier, would one of these work in lieu of doing a video or call? I’m open to whatever is easiest 

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

https://gist.github.com/CristinaSolana/1885435

**Comment by serundeputy** [comment]
this one works https://gist.github.com/CristinaSolana/1885435

**Pull request closed without merging**